ObjectiveCloudant/Operations/CDTQueryViewOperation.m

```diff
@@ -0,0 +1,295 @@
+//
+#import "CDTQueryViewOperation.h"
+
+static const int kCDTDefaultOperationIntegerValue = -1;
+
+@implementation CDTQueryViewOperation
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _descending = NO;
+        _group = NO;
+        _groupLevel = kCDTDefaultOperationIntegerValue;
+        _includeDocs = NO;
+        _inclusiveEnd = NO;
+        _reduce = NO;
+        _limit = kCDTDefaultOperationIntegerValue;
+        _skip = kCDTDefaultOperationIntegerValue;
+        _stale = CDTStaleViewNo;
+    }
+    return self;
+}
+
+- (BOOL)buildAndValidate
+{
+    if (![super buildAndValidate]) {
+        return NO;
+    }
+
+    if (!self.ddoc) {
+        return NO;
+    }
+
+    if (!self.viewName) {
+        return NO;
+    }
+
+    // Query params
+
+    if (self.endKey && ![CDTQueryViewOperation isValidStringOrJsonArray:self.endKey]) {
+        return NO;
+    }
+
+    if (self.endkeyDocId && ![CDTQueryViewOperation isValidStringOrJsonArray:self.endkeyDocId]) {
+        return NO;
+    }
+
+    if (self.key && self.keys) {
+        return NO;
+    }
+
+    if (self.key && ![CDTQueryViewOperation isValidStringOrJsonArray:self.key]) {
+        return NO;
+    }
+
+    if (self.keys && ![CDTQueryViewOperation isValidStringOrJsonArray:self.keys]) {
+        return NO;
+    }
+
+    if (self.startKey && ![CDTQueryViewOperation isValidStringOrJsonArray:self.startKey]) {
+        return NO;
+    }
+
+    if (self.startKeyDocId &&
+        ![CDTQueryViewOperation isValidStringOrJsonArray:self.startKeyDocId]) {
+        return NO;
+    }
+
+    return YES;
+}
+
+- (NSString *)httpMethod { return @"GET"; }
+- (NSString *)httpPath
+{
+    return [NSString
+        stringWithFormat:@"/%@/_design/%@/_view/%@", self.databaseName, self.ddoc, self.viewName];
+}
+
+- (nonnull NSArray<NSURLQueryItem *> *)queryItems
+{
+    NSMutableArray<NSURLQueryItem *> *queryItems = [NSMutableArray array];
+
+    [queryItems addObject:[NSURLQueryItem queryItemWithName:@"descending"
+                                                      value:[CDTQueryViewOperation
+                                                                stringForBoolean:self.descending]]];
+    if (self.endKey) {
+        [queryItems addObject:[NSURLQueryItem queryItemWithName:@"endkey"
+                                                          value:[CDTQueryViewOperation
+                                                                    stringForObject:self.endKey]]];
+    }
+
+    if (self.endkeyDocId) {
+        [queryItems
+            addObject:[NSURLQueryItem queryItemWithName:@"endkey_docid" value:self.endkeyDocId]];
+    }
+
+    [queryItems addObject:[NSURLQueryItem queryItemWithName:@"group"
+                                                      value:[CDTQueryViewOperation
+                                                                stringForBoolean:self.group]]];
+
+    if (self.groupLevel != kCDTDefaultOperationIntegerValue) {
```

Comment:
Bad name given this is specific to this operation. Should be something like `kCDTDefaultQueryViewOperationIntegerValue` I guess :/